### PR TITLE
feat(server): include file creation date in the list

### PR DESCRIPTION
**Pull request by viktaur**
## Description
This PR includes the date and time in ISO8601 format of the instant the file was created.

## Motivation and Context

The information returned in the list function was missing a timestamp representing the date of creation of the file, which can be useful for sorting or retrieving the latest files.

This fixes issue #293.
## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (no code change)
- [x] Refactor (refactoring production code)
- [ ] Other 

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly.
- [ ] I have formatted the code with [rustfmt](https://github.com/rust-lang/rustfmt).
- [x] I checked the lints with [clippy](https://github.com/rust-lang/rust-clippy).
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -49,6 +49,7 @@ tracing = "0.1.40"
 tracing-subscriber = { version = "0.3.18", features = ["env-filter"] }
 uts2ts = "0.4.1"
 path-clean = "1.0.1"
+chrono = "0.4.38"
```
Comment (orhun) [inline Cargo.toml:52]: Is this dependency being used?

**Review by orhun — COMMENTED** [review]
ping @tessus for additional review

File: Cargo.toml (hunk 1 above)
Comment (viktaur) [reply Cargo.toml:52]: It was not used in the end. Forgot to remove it, sorry.

Comment (tessus) [reply Cargo.toml:52]: Sorry, I was busy with somwthing else. No, `chrono` has not been and is not used. It was one of the reasons why I created the uts2ts crate. ;-)

File: src/server.rs [hunk 2]
```diff
@@ -369,11 +390,14 @@ async fn list(config: web::Data<RwLock<Config>>) -> Result<HttpResponse, Error>
...
-                Some(ListItem {
-                    file_name,
-                    file_size: metadata.len(),
-                    expires_at_utc,
-                })
+                Some(
+                    ListItem {
+                        file_name,
+                        file_size: metadata.len(),
+                        creation_date,
```
Comment (tessus) [inline src/server.rs:397]: the uts2ts crate is not TZ aware, thus all timestamps are UTC.
TZ calculations are complex and we decided back then not to use the `time` or `chrono` crate because they pull in a lot of dependencies.

thus this should be named consistently with the `expires_at_utc`:

```suggestion
                        creation_date_utc,
```

File: src/server.rs [hunk 3]
```diff
@@ -322,6 +322,8 @@ pub struct ListItem {
     pub file_name: PathBuf,
     /// Size of the file in bytes.
     pub file_size: u64,
+    /// ISO8601 formatted date-time of the moment the file was created (uploaded).
+    pub creation_date: Option<String>,
```
Comment (tessus) [inline src/server.rs:326]:
```suggestion
    pub creation_date_utc: Option<String>,
```

File: src/server.rs [hunk 4]
```diff
@@ -354,6 +356,25 @@ async fn list(config: web::Data<RwLock<Config>>) -> Result<HttpResponse, Error>
                     }
                 };
                 let mut file_name = PathBuf::from(e.file_name());
+
+                let creation_date = match metadata
```
Comment (tessus) [inline src/server.rs:360]:
```suggestion
                let creation_date_utc = match metadata
```

**Comment by tessus** [comment]
We also need another PR for the rustypaste-cli. ;-)

**Review by tessus — COMMENTED** [review]
LGTM. Thanks for this PR.

File: Cargo.lock
Comment (orhun) [inline Cargo.lock:1]: Can you revert these changes?

Comment (tessus) [reply Cargo.lock:1]: oops. I missed that this file was still there, since I checked the `viewed` box...
I guess it is still a leftover from the change to the cargo.toml.

Comment (viktaur) [reply Cargo.lock:1]: I have removed chrono, wasm-bindgen, and js-sys from dependencies. Does that work or do you want me to replace Cargo.lock with the version before the PR?

Comment (orhun) [reply Cargo.lock:1]: Let's not touch `Cargo.lock` in this PR 🐻 would be nice to revert the changes.

File: src/server.rs [hunk 5]
```diff
@@ -354,6 +356,25 @@ async fn list(config: web::Data<RwLock<Config>>) -> Result<HttpResponse, Error>
...
+                    Ok(millis) => Some(
+                        uts2ts::uts2ts(
+                            i64::try_from(millis)
+                                .expect("UNIX time should be smaller than i64::MAX")
+                            / 1000
+                        )
+                        .as_string()
+                    ),
+                    Err(_) => None
+                };
```
Comment (orhun) [inline src/server.rs:376]: If we are going to ignore the error then this can be refactored into:

```suggestion
                let creation_date_utc = metadata
                    .created()
                    .ok()
                    .map(|v| {
                        v.duration_since(UNIX_EPOCH)
                            .expect("Time since UNIX epoch should be valid.")
                            .as_millis()
                    })
                    .map(|millis| {
                        uts2ts::uts2ts(
                            i64::try_from(millis)
                                .expect("UNIX time should be smaller than i64::MAX")
                                / 1000,
                        )
                        .as_string()
                    });
```

File: Cargo.lock
[the remark this answers is not in the document]
Comment (viktaur) [reply Cargo.lock:1]: Hi, the proposed Cargo.lock is up to date with the current one, where `libc` has version 0.2.155. Unless you want me to revert it to version 0.2.153, as it was at the moment the PR was started.

**Comment by orhun** [comment]
Can you run `cargo fmt` and commit the changes pls 😇

**Review by orhun — APPROVED** [review]
Thanks for your contribution!
